homeassistant/components/weather/__init__.py

```diff
@@ -110,7 +110,8 @@ def state_attributes(self):
             ATTR_WEATHER_TEMPERATURE: show_temp(
                 self.hass, self.temperature, self.temperature_unit,
                 self.precision),
-            ATTR_WEATHER_HUMIDITY: round(self.humidity)
+            ATTR_WEATHER_HUMIDITY: (round(self.humidity) if self.humidity
```

Comment:
The `if self.humidity` check will not work with falsy values like `0.0`.

Write something like `if self.humidity is not None` or, even better, make it an additional check like the other attributes:

```python
data = {
            ATTR_WEATHER_TEMPERATURE: show_temp(
                self.hass, self.temperature, self.temperature_unit,
                self.precision),
        }

        humidity = self.humidity
        if humidity is not None:
            data[ATTR_WEATHER_HUMIDITY] = humidity
```

Reply:
You are absolutely right, hadn't thought about that.
Even though humidity is probably never going to be 0, we should definitely take it into account.
Fixed in the commit below.